extensions/kubernetes-client/runtime-internal/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientUtils.java

```diff
@@ -93,4 +123,15 @@ public static KubernetesClient createClient() {
                 .build())
                 .build();
     }
+
+    public static Set<Class<? extends KubernetesResource>> scanKubernetesResources() {
```

Comment:
I am pretty sure this can be done at build time and the mapper proper populated? Would you like me to take a look?

Reply:
I'll have a look at it tomorrow morning

Reply:
We discussed using Jandex for this registration at some point, looks like it would be really useful here.

Reply:
All of the jars provided by the fabric8 project include a jandex index.  We only decided against using those in the project to avoid the jandex dependency.

Reply:
If we do have Jandex indices, though, we could use Jandex here instead of class path scanning, no?

Reply:
We switched from jandex to SPI. The fact that SPI is not working is the main problem. Probably by making some build time output we can overcome this static issue (waiting on geoand take on this)

Reply:
That's on the registration side of things, though, right? If the fabric8 build produces indices, we could leverage them here. That said, that would indeed require that people add the jandex plugin to their build when they add new classes. That's one aspect of the client that I'm not too familiar with, actually: who is supposed to use that SPI feature?